### PR TITLE
[wip] Bandwidth mc command

### PR DESCRIPTION
**Pull request by BigUstad**
`mc admin bandwidth --buckets <bucket1> --buckets <bucket2> <mino-server>` for a single server setup
`mc admin bandwidth --buckets <bucket1> --buckets <bucket2> --servers <server1> --servers <server2> --servers <server3> --servers <server4>` for a distributed setup

### CODE REVIEW

File: cmd/admin-bw-info.go [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+	"github.com/minio/cli"
+	json "github.com/minio/mc/pkg/colorjson"
+	"github.com/minio/mc/pkg/probe"
+	"github.com/minio/minio/pkg/bucket/bandwidth"
+	"github.com/minio/minio/pkg/console"
+	"github.com/minio/minio/pkg/madmin"
+)
+
+var adminBwInfoCmd = cli.Command{
+	Name:   "bandwidth",
```
Comment (kerneltime) [inline cmd/admin-bw-info.go:33]: The file name should be `admin-bandwidth-info.go` will make it easier to find

File: cmd/admin-bw-info.go [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+}
+
+var adminBwFlags = []cli.Flag{
+	cli.StringSliceFlag{
+		Name:  "buckets",
+		Usage: "format '--buckets <bucket>', multiple values allowed for multiple buckets",
+	},
+	cli.StringSliceFlag{
+		Name:  "servers",
+		Usage: "format '--servers <server>', multiple values allowed for multiple servers for a distributed setup",
```
Comment (kerneltime) [inline cmd/admin-bw-info.go:62]: The Admin API should be querying all the servers and the mc command only needs to talk to any one of the instances.

File: cmd/admin-bw-info.go [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+  1. Show the bandwidth usage for the MinIO server
+     {{.Prompt}} {{.HelpName}} --buckets bucket1 --buckets bucket2 play/
+  2. Show the bandwidth usage for a distributed MinIO servers setup
+     {{.Prompt}} {{.HelpName}} --buckets bucket1 --buckets bucket2 --servers minio01/ --servers minio02/ --servers minio03/ --servers minio04/
+`,
+}
+
+var adminBwFlags = []cli.Flag{
+	cli.StringSliceFlag{
+		Name:  "buckets",
```
Comment (kerneltime) [inline cmd/admin-bw-info.go:57]:
```suggestion
		Name:  "bucket",
```

File: cmd/admin-bw-info.go [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+
+USAGE:
+  {{.HelpName}} TARGET
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+  1. Show the bandwidth usage for the MinIO server
+     {{.Prompt}} {{.HelpName}} --buckets bucket1 --buckets bucket2 play/
```
Comment (kerneltime) [inline cmd/admin-bw-info.go:49]: Help documentation needs updating.

File: cmd/admin-bw-info.go [hunk 5]
```diff
@@ -0,0 +1,392 @@
...
+	console.SetColor("InfoFail", color.New(color.FgRed, color.Bold))
+	console.SetColor("BlinkLoad", color.New(color.BlinkSlow, color.FgCyan))
+	msg += fmt.Sprintf("%s  %s\n", console.Colorize("Info", dot), console.Colorize("PrintB", b.Server))
+	for _, bucket := range b.Buckets {
+		sumAg := fmt.Sprintf("%.4f", b.Info.SampleResult[bucket].SumAggregateBandwidth)
+		sumMv := fmt.Sprintf("%.4f", b.Info.SampleResult[bucket].SumMovingBandwidth)
+		avgAg := fmt.Sprintf("%.4f", b.Info.SampleResult[bucket].AvgAggregateBandwidth)
+		avgMv := fmt.Sprintf("%.4f", b.Info.SampleResult[bucket].AvgMovingBandwidth)
+		msg += fmt.Sprintf("   Bucket: %s\n", console.Colorize("Info", bucket))
+		msg += fmt.Sprintf("      Sum Aggregate Bandwidth  : %s\n", console.Colorize("Info", sumAg))
```
Comment (kerneltime) [inline cmd/admin-bw-info.go:91]: This needs clean up, we don't have these values any more.

**Comment by kerneltime** [comment]
I don't think we should loop when printing json

**Comment by kerneltime** [comment]
Done testing the cli

**Comment by BigUstad** [comment]
Implemented at https://github.com/minio/mc/pull/3436

**Pull request closed without merging**